### PR TITLE
Upgrade gibbon

### PR DESCRIPTION
**Pull request by TheDudeWithTheThing**
Give this a go. I haven't actually tested it yet so be sure to do that before pushing it.

### CODE REVIEW

File: app/models/mail_chimp_client.rb [hunk 1]
```diff
@@ -5,63 +5,49 @@ class MailChimpClient
...
     merge_vars = {
       'FNAME' => name
     }
 
-    gibbon.lists.subscribe(
-      id: USER_LIST_ID,
-      email: { email: email },
-      merge_vars: merge_vars,
-      double_optin: true,
-      send_welcome: false
```
Comment (markottaviani) [inline app/models/mail_chimp_client.rb]: How is the ```double_optin``` and ```send_welcome``` handled now?  Don't think ```double_optin``` is working right now.

Comment (TheDudeWithTheThing) [inline app/models/mail_chimp_client.rb]: From what I gathered from the docs they are settings on the list not Params you pass up in the subscribe call. I don't think I can see the prod lists so you'll have to poke around

File: app/models/mail_chimp_client.rb [hunk 2]
```diff
@@ -5,63 +5,49 @@ class MailChimpClient
...
-    @gibbon = Gibbon::API.new
+    @gibbon = Gibbon::Request.new
     @gibbon.timeout = 30
   end
 
+  def subscribe(list_id, email, merge_vars)
+    gibbon.lists(list_id).members(hashed(email)).upsert(
+      body: {
+        email_address: email,
+        status: 'subscribed',
```
Comment (markottaviani) [inline app/models/mail_chimp_client.rb]: Found it, looks like status should be set to ```pending``` here, and that will trigger the double-opt in process.  

https://pbjmarketing.com/easily-disable-mailchimp-double-opt-in-in-api-3-0/

**Comment by markottaviani** [comment]
user sub and unsub seem good.  

Hitting errors on the adopter subscribe:

```
the server responded with status 400 @title="Invalid Resource", @detail="The resource submitted could not be validated. For field-specific details, see the 'errors' array.", @body={"type"=>"http://developer.mailchimp.com/documentation/mailchimp/guides/error-glossary/", "title"=>"Invalid Resource", "status"=>400, "detail"=>"The resource submitted could not be validated. For field-specific details, see the 'errors' array.", "instance"=>"", "errors"=>[{"field"=>"merge_fields.groupings", "message"=>"Data did not match any of the schemas described in anyOf."}]}, @raw_body="{\"type\":\"http://developer.mailchimp.com/documentation/mailchimp/guides/error-glossary/\",\"title\":\"Invalid Resource\",\"status\":400,\"detail\":\"The resource submitted could not be validated. For field-specific details, see the 'errors' array.\",\"instance\":\"\",\"errors\":[{\"field\":\"merge_fields.groupings\",\"message\":\"Data did not match any of the schemas described in anyOf.\"}]}", @status_code=400
/Users/mark/.rvm/gems/ruby-2.2.3/gems/gibbon-2.2.4/lib/gibbon/api_request.rb:120:in `handle_error'
/Users/mark/.rvm/gems/ruby-2.2.3/gems/gibbon-2.2.4/lib/gibbon/api_request.rb:42:in `rescue in put'
/Users/mark/.rvm/gems/ruby-2.2.3/gems/gibbon-2.2.4/lib/gibbon/api_request.rb:36:in `put'
/Users/mark/.rvm/gems/ruby-2.2.3/gems/gibbon-2.2.4/lib/gibbon/request.rb:51:in `upsert'
/Users/mark/rails_projects/RescueRails/app/models/mail_chimp_client.rb:13:in `subscribe'
/Users/mark/rails_projects/RescueRails/app/models/mail_chimp_client.rb:39:in `adopter_subscribe'
/Users/mark/rails_projects/RescueRails/app/services/mail_chimp_service.rb:18:in `adopter_subscribe'
/Users/mark/rails_projects/RescueRails/app/jobs/adopter_subscribe_job.rb:5:in `perform'
/Users/mark/.rvm/gems/ruby-2.2.3/gems/activejob-4.2.6/lib/active_job/execution.rb:32:in `block in perform_now'
```

**Comment by TheDudeWithTheThing** [comment]
can you test with master branch first?  I think the dev version of the adopter list might not be set up the same way production is, I can't see prod though so it's all you.

https://us8.admin.mailchimp.com/lists/dashboard/groups?id=593717

**Comment by TheDudeWithTheThing** [comment]
wrong link, this is where the merge vars are defined, I'm guessing you have to define the groupings one?

https://us8.admin.mailchimp.com/lists/settings/merge-tags?id=593717

**Comment by TheDudeWithTheThing** [comment]
@markottaviani any word here?

**Comment by markottaviani** [comment]
It's been a crazy week, I should be able to look at this by tomorrow morning a the latest.

**Comment by markottaviani** [comment]
@TheDudeWithTheThing   okay, the Test mailchimp account has been updated.  the two lists there line up in terms of data fields and group information.  

My prior testing was against the prod mailchimp that hit the errors.

Thanks again for your help!

**Comment by markottaviani** [comment]
Okay, I think the errors are related to the groupings/groups/interests which are apparently the same thing in Mailchimp.  

Pushed some TODO notes of where I think the changes are needed, though I'm not quite sure how to implement this yet.

**Comment by markottaviani** [comment]
- [ ] Will need to query Mailchimp to determine if subscribe or update action should be used.  (Need to use ```status: 'pending' ``` to make sure peple still get email confirmations.  
- [ ] Make sure user isn't emailed about resubscription each time mailchimp info might need to be updated.  
- [ ] Need to deal with interests in user Subscribe
- [ ] Move Interests and List IDs to ENV params

**Comment by TheDudeWithTheThing** [comment]
K will hit this up on the train in tomorrow. For some reason github didn't send me notifications for the message 8 days ago

**Comment by TheDudeWithTheThing** [comment]
>  Will need to query Mailchimp to determine if subscribe or update action should be used. (Need to use status: 'pending' to make sure peple still get email confirmations.

upsert doesn't handle this for us?

**Comment by TheDudeWithTheThing** [comment]
> Move Interests and List IDs to ENV params

I'm going to pull this out to a different PR

**Comment by markottaviani** [comment]
I think the issue with upsert is the ```status: pending``` .

Right now after someone submits an application they get added to Mailchimp with the status: pending.  Then they opt-in to mailchimp via mailchimp opt-in email and then the status on mailchimp changes to subscribed.  

If the status of their adoption application changes and we want to add an interest in Mailchimp, they get set back to ```status: pending``` again and get another confirmation email.  :disappointed: 

I'd be curious to see what happens if we remove ```status:``` all together.  If it defaults to pending for a new subscribe in mailchimp then we might be golden.  

Otherwise we'll need to write an ```update()``` method for the ```mail_chimp_client.rb``` that gets called when someone's Adoption Application changes.

**Comment by TheDudeWithTheThing** [comment]
huge

@detail="shaun@sbutler.org needs a 'status' or 'status_if_new' field included

**Comment by markottaviani** [comment]
Anything else you wanted to do with this, or should I run through some tests and push it out?

**Comment by TheDudeWithTheThing** [comment]
I think it's good to test? The test env for mailchimp didn't seem very useful running locally so not sure what else to do

**Comment by markottaviani** [comment]
When trying to add a new User via user_subscribe I get an invalid Interest error message because the Adoption Interests are still getting added to the request.  So it seems that interests for other lists can't be mentioned at all.    Attached a commit that fixes the issues.

**Comment by markottaviani** [comment]
Okay I think this is good to go.  Tests are working well.  Check out the changes I made and let me know if you have any other tweaks or if it's good to go

**Comment by TheDudeWithTheThing** [comment]
Seems good!

**Pull request merged**